hw/top_earlgrey/data/top_earlgrey.hjson

```diff
@@ -445,6 +467,13 @@
           type: "req_rsp"
           act: "rsp"
           name: "tl"
+        },
+        // Interface to SRAM controller
```

Comment:
ah are you still considering wrapping this up inside the memory controller?
FWIW, i think it's fine to leave it like this also.  The wrapping up is a nice optimization, but not critical IMO.

Reply:
Actually I had it wrapped up before, but top-level integration is more straight-forward if the memory macro is separate. I.e., it is easier to declare and wire up the memory TL-UL port in topgen if the memory.

Hence I am pulling the memory primitive out of `sram_ctrl` in this PR.

Reply:
yeah that sounds fine to me.
We can wrap up later (flash, all the memories) later as a different optimization. 

As a separate note, say I wanted to add another small memory for purposes....but would like to actually share the scrambler / de-scrambler of one of the main srams, do you think there's an easy way to add that enhancement?

Reply:
hm... that would require separating the scrambling mechanism and memory instance. at the moment, both are combined in `prim_ram_1p_scr`, which from the outside just looks like the other `ram_1p` primitives with an additional key input. I structured it like that for convenience mostly, such that the scrambling device can be conveniently instantiated in OTBN.

do you have a specific use case in mind where we would need to share the scrambling logic among several memories?

Reply:
this is mostly for the tightly-copuled-execution memory usecase for the host processor..
it's likely we'll need a piece of memory where they can execute code.  Ideally, this should be a pmp protected region within the main sram, but there is SOME argument for making this a physically separate RAM. 

But to be clear, I don't think we have to make the ram controller aware of it right? We could build some glue logic outside around the sram adapter basically to know there are two physical instances.

Reply:
Yeah the structural changes I mentioned above can be done outside of `sram_ctrl`, I think.

One question though... this is meant to be used in test-mode only, right?
Is there a large benefit in adding an extra memory for this or could we just modulate the TL-UL connection from icache to the main SRAM with LC_DFT_EN?

Reply:
ah that's the problem... this is going to be during production mode :(
which is why we're thinking about whether a physically separate instance
makes sense. It comes down to how much protection we can build around a
particular section of memory.

Reply:
ugh... ok I see what you mean.

Reply:
there's no decision here yet... so don't worry about it too much yet.
But wanted to get it on your radar.

Reply:
ok, thanks - I took note.